kornia/feature/sold2/sold2_detector.py

```diff
@@ -61,25 +100,24 @@ class SOLD2_detector(Module):
         >>> line_segments = sold2_detector(img)["line_segments"]
     """
 
-    def __init__(self, pretrained: bool = True, config: Optional[Dict[str, Any]] = None) -> None:
+    def __init__(self, pretrained: bool = True, config: Optional[DetectorCfg] = None) -> None:
```

Comment:
I think it's ok if we drop support for the dict as config on the typing already, but over the functionality we should keep support with a warning, something like

`import warnings`
```suggestion
    def __init__(self, pretrained: bool = True, config: Optional[DetectorCfg] = None) -> None:
        if isinstance(cofig, dict):
            warnings.warn(
                    f"Usage of config as a plain dictionary is deprecated in favor of `kornia.feature.sold2.sold2_detector.DetectorCfg`. The support of plain dictionaries as config will be removed in kornia v0.8.0 (December 2024).",
                    category=DeprecationWarning,
                    stacklevel=2,
                )
            config = dict_to_dataclass(**config, DetectorCfg)
```